scipy/stats/_rvs_sampling.py

```diff
@@ -128,13 +132,6 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
     if umax <= 0:
         raise ValueError("umax must be positive.")
 
-    exp_iter = 2 * (vmax - vmin) * umax  # rejection constant (see [1])
-    if exp_iter > 20:
-        msg = ("The expected number of iterations to generate a single random "
-               "number from the desired distribution is larger than {}, "
-               "potentially causing bad performance.".format(int(exp_iter)))
-        warnings.warn(msg, RuntimeWarning)
-
```

Comment:
@chrisb83 I'm a little confused here, I always thought that the pdf could be specified up to a proportionality constant? 

I also always assumed that the `exp_iter > 20` would come from a bounding rectangle that poorly covers the region A that you sample. 

Even with the changes in this PR, you could still have a poorly approximated ratio, area(A)/area(R), that is passed into the RoU method. 

Or did I miss something?

Reply:
yes, the input is a function proportional to the pdf. You cannot check the the expected number of iterations unless it is the pdf since you do not know the integral of f. so the warning could be incorrectly raised. we still have an exception if no rvs can be generated

Reply:
ok, got it, thanks for clarifying.